### PR TITLE
vapoursynth: Add version 57

### LINKED ISSUE #3229: [Request] VapourSynth
### Criteria

For a package to be acceptable in the Main bucket, it should be:

- [X] a non-GUI tool
- [X] a reasonably well-known and widely used developer tool
- [X] the latest stable version of the program
- [X] the full version i.e. not a trial version
- [X] a fairly standard install (e.g. uses a version-specific download URL, no elaborate pre/post install scripts)

### Information

**Name:** vapoursynth

**Description:** A video processing framework with simplicity in mind

**Homepage:** http://www.vapoursynth.com

**Download link(s):** https://github.com/vapoursynth/vapoursynth/releases/download/R57/VapourSynth64-Portable-R57.7z

**Some indication of popularity/repute:** https://github.com/vapoursynth/vapoursynth

### PR DESCRIPTION
**Pull request by couleurm**
Closes #3229

- [X] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by couleurm** [comment]
How should I do different persist folders depending on architecture?

**Comment by issaclin32** [comment]
> How should I do different persist folders depending on architecture?

Unfortunately Scoop does not provide `persist` for architecture block.
You can use **pre_install** and **uninstaller -> script** to do manual persist.

such as:
```
    "pre_install": [
        "$arch = $architecture.SubString(0,2)",
        "if(Test-Path \"$persist_dir\\vapoursynth$arch\") {Copy-Item \"$persist_dir\\*\" \"$dir\\\" -Force -Recurse}"
    ],
    "uninstaller": {
        "script": [
            "$arch = $architecture.SubString(0,2)",
            "ensure \"$persist_dir\" | Out-Null",
            "Copy-Item \"$dir\\vapoursynth$arch\" \"$persist_dir\\\" -Force -Recurse"
        ]
    },
```

`$persist_dir` will not exist if your script does not have `persist` field, therefore we need `ensure` to create the folder first.

File: bucket/vapoursynth.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{
+    "version": "R57",
+    "description": "A video processing framework with simplicity in mind ",
+    "homepage": "http://www.vapoursynth.com/",
+    "depends": [
```
Comment (issaclin32) [inline bucket/vapoursynth.json:5]:
```suggestion
    "depends": "extras/vcredist2022"
```
**VCRedist2022** now covers all the functions from VC2015-2022.
https://docs.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist?view=msvc-170

File: bucket/vapoursynth.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
+{
+    "version": "R57",
```
Comment (rashil2000) [inline bucket/vapoursynth.json:2]: `57` is sufficient. The `R` prefix is not needed. Please modify checkver and autoupdate accordingly.

File: bucket/vapoursynth.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    "description": "A video processing framework with simplicity in mind ",
+    "homepage": "http://www.vapoursynth.com/",
```
Comment (rashil2000) [inline bucket/vapoursynth.json:4]:
```suggestion
    "homepage": "https://www.vapoursynth.com",
```

File: bucket/vapoursynth.json [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    "depends": [
+        "https://github.com/ScoopInstaller/Extras/raw/master/bucket/vcredist2019.json"
+    ],
+    "license": {
+        "identifier": " LGPL-2.1",
+        "url": "https://github.com/vapoursynth/vapoursynth/blob/master/COPYING.LESSER"
+    },
```
Comment (rashil2000) [inline bucket/vapoursynth.json:11]:
```suggestion
    "license": "LGPL-2.1-or-later",
```

File: bucket/vapoursynth.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        "AVFS.exe",
+        "VSPipe.exe",
+        "pfm-192-vapoursynth-win.exe",
+        "vsrepo.py"
+    ],
+    "post_install": [
+        "if (-Not(Get-Command -Name python.exe -ErrorAction Ignore)){",
+        "$shimsdir = Join-Path $scoopdir shims",
+        "@('.exe','.shim') | ForEach-Object {Copy-Item \"$shimsdir\\python39$_\" \"$shimsdir\\python$_\"}}"
+    ],
```
Comment (rashil2000) [inline bucket/vapoursynth.json:36]:
```suggestion
```

This is not required. versions/python39 does this automatically.

**Review by rashil2000 — CHANGES_REQUESTED** [review]

File: bucket/vapoursynth.json (hunk 5 above)
Comment (couleurm) [reply bucket/vapoursynth.json:33]: It only makes a shim for python3 and 39 https://github.com/ScoopInstaller/Versions/blob/master/bucket/python39.json

**Review by rashil2000 — APPROVED** [review]
There are spaces at the end of notes and description.

**Review by issaclin32 — APPROVED** [review]

**Pull request merged**